src/Marten/Linq/Parsing/SimpleBinaryComparisonExpressionParser.cs

```diff
@@ -34,29 +34,30 @@ public bool Matches(BinaryExpression expression)
 
         public IWhereFragment Parse(IQueryableDocument mapping, ISerializer serializer, BinaryExpression expression)
         {
-            var isValueExpressionOnRight = expression.Right.IsValueExpression();
+            var areBothMemberExpresions = !expression.Left.IsValueExpression() && !expression.Right.IsValueExpression();
```

Comment:
Please rename the variable from `areBothMemberExpresions` to `areBothMemberExpressions`

Reply:
Fixed